Userland/Libraries/LibGfx/ImageFormats/JPEGLoader.cpp

```diff
@@ -1124,6 +1129,24 @@ static inline void set_macroblock_metadata(JPEGLoadingContext& context)
     context.mblock_meta.total = context.mblock_meta.hcount * context.mblock_meta.vcount;
 }
 
+static ErrorOr<void> ensure_standard_precision(StartOfFrame const& frame)
+{
+    // B.2.2 - Frame header syntax
+    // Table B.2 - Frame header parameter sizes and values
+
+    if (frame.type == StartOfFrame::FrameType::Baseline_DCT && frame.precision == 8)
+        return {};
+
+    if (frame.type == StartOfFrame::FrameType::Extended_Sequential_DCT && frame.precision == 8)
+        return {};
+
+    if (frame.type == StartOfFrame::FrameType::Progressive_DCT && frame.precision == 8)
+        return {};
```

Comment:
Is `if (frame.precision == 8) return {};` not always correct? (Instead of the first 3 explicit case.)